src/asm/symbol.c

```diff
@@ -533,42 +533,51 @@ static struct Symbol *addLabel(char const *symName)
 }
 
 /*
- * Add a local (.name or Parent.name) relocatable symbol
+ * Add a local (`.name` or `Parent.name`) relocatable symbol
  */
 struct Symbol *sym_AddLocalLabel(char const *symName)
 {
 	if (!labelScope) {
 		error("Local label '%s' in main scope\n", symName);
 		return NULL;
 	}
+	assert(!strchr(labelScope, '.')); /* Assuming no dots in `labelScope` */
+
+	char fullName[MAXSYMLEN + 1];
+	char const *localName = strchr(symName, '.');
+
+	assert(localName); /* There should be at least one dot in `symName` */
+	assert(localName[0] == '.'); /* `localName` should start with a dot */
```

Comment:
This is guaranteed by the interface of `strchr` as long as `localName` is non-NULL, so this can be removed.